webpack/components/Search/Search.js

```diff
@@ -38,74 +39,55 @@ class Search extends Component {
       params.params || {},
     ];
 
-    if (autoCompleteParams[0] !== '') {
+    if (params.endpoint !== '') {
       let data;
-      if (this.props.foremanApiAutoComplete) {
+      if (foremanApiAutoComplete) {
         data = await foremanApi.get(...autoCompleteParams);
       } else {
         data = await api.get(...autoCompleteParams);
       }
-      this.setState({
-        items: data?.data?.filter(({ error }) => !error).map(({ label }) => ({
-          text: label.trim(),
-        })),
-      });
+      setItems(data?.data?.filter(({ error }) => !error).map(({ label }) => ({
+        text: label.trim(),
+      })));
     }
 
-    if (autoSearchEnabled && patternfly4 && searchTerm.length > 0) {
-      this.autoSearch(searchTerm);
+    if (autoSearchEnabled && patternfly4) {
+      onSearch(searchTerm || '');
     }
   };
 
-  onSearch = (search) => {
-    if (this.props.updateSearchQuery) this.props.updateSearchQuery(search);
-    this.props.onSearch(search);
-  };
+  useEffect(() => {
+    loadSetting(AUTOSEARCH_DELAY);
+    loadSetting(AUTOSEARCH_WHILE_TYPING);
+    // This is needed to prevent cross-repo hook dependencies.
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
I see `loadSetting` is defined in Foreman like this:
```js
export const loadSetting = settingName => async dispatch => {
  dispatch({ type: GET_SETTING_REQUEST });
  try {
    const { data } = await API.get(`/api/v2/settings/${settingName}`);
    return dispatch({
      type: GET_SETTING_SUCCESS,
      response: data,
    });
  } catch (error) {
    return dispatch({
      type: GET_SETTING_FAILURE,
      result: error,
    });
  }
};
```

* I'm unclear on why it's being passed down here as a prop, rather than simply imported from `foremanReact`.
* Also, it's using our old action style of thunk dispatch -> await -> success or failure dispatch.  We've moved on from that and I think our API middleware pattern is much better.

With that said, what are your thoughts on either (a) importing it directly as-is, or (b) writing a new action here in Katello, the modern way?  Or even (c) fixing the one in Foreman :)  I think any one of these options would remove the need to disable the rule here.

Reply:
I chose the first option, worked well.